2023/incoming/santa-tackles-bitcoin-part-two.pod

```diff
@@ -0,0 +1,126 @@
+
+# how many presents?
+my $number_of_kids = 346;
+
+# print all the keys in WIF format
+foreach my $index (0 .. $number_of_kids - 1) {
+
+	# same derivation scheme as before
+	my $key = $master_key->derive_key_bip44(
+		purpose => 84,
+		account => 0,
+		index => $index,
+	)->get_basic_key;
+
+	# print it to standard output
+	say $key->to_wif;
+}
+
+=end perl
+
+This script was much more straightforward than the previous one! It was time to
+see if the key recovery worked:
+
+	Please provide the mnemonic and password, each in separate lines:
+	brown bulk hire culture capital hill trim turkey  gossip artefact door media argue basic execute slam minute try number daughter music gauge vocal wink
+	Christmas
+	L1pn6bRKUY1dMXg6XvURHNCrm9CDByFehQPHmnfTKQXCxRktUAqN
+	L3dhFVe8BgjgGGbu2Rm72fcgyB3SgVZ2ee8fsoezSA47QxhDAng5
+	L2TZEBZJ2yTb2JxcxbUippr3DBAkZopLFrXSGGZRRzfH78RFcTnj
+	(...)
+
+It worked indeed, but it was not yet clear whether the private keys were the
+correct ones. McJingles crafted a one-liner to test the first key he got:
+
+	$ perl -MBitcoin::Crypto=btc_prv -e \
+	'print btc_prv->from_wif("L1pn6bRKUY1dMXg6XvURHNCrm9CDByFehQPHmnfTKQXCxRktUAqN")->get_public_key->get_address'
+	bc1qls9ffahaa5kezk0zvq2hh58nn6yfv8uehdq7dp
+
+Something was wrong. That's not the address which was generated before! How is
+that possible? After thorough investigation McJingles found out he typed two
+spaces while providing the mnemonic for the program. A quick scan through
+L<Bitcoin::Crypto::Key::ExtPrivate>'s documentation has revealed that
+C<from_mnemonic> method usage without the language argument is not only
+whitespace-sensitive but also doesn't check if the words exist in the
+dictionary. It is pretty unsafe to use it like that unless you nail it exactly
+right! He modified the script to add C<'en'> language as the last argument to
+C<from_mnemonic>...
+
+=begin perl
+
+my $master_key = btc_extprv->from_mnemonic($mnemonic, $password, 'en');
+
+=end perl
+
+... and then tried again, making the same mistake:
+
+	Please provide the mnemonic and password, each in separate lines:
+	brown bulk hire culture capital hill trim turkey  gossip artefact door media argue basic execute slam minute try number daughter music gauge vocal wink
+	Christmas
+	L4DrMfpZRp5P6szTGqWQHnYDBJaAwTRN8NHRkTM3Kc79U14wi5Li
+	L3rpcJdvNynUCWWaQEEAXkS7FPpCrz1HnCiHT4VYcmm4oWct5Fd8
+	KwzsywmntPwwnEMTbd9jw7wPjeM4PKx52KeSLU9cZ4o1Qx7AFyTD
+	(...)
+
+The output was different, but was it the right key this time?
+
+	$ perl -MBitcoin::Crypto=btc_prv -e \
+	'print btc_prv->from_wif("L4DrMfpZRp5P6szTGqWQHnYDBJaAwTRN8NHRkTM3Kc79U14wi5Li")->get_public_key->get_address'
+	bc1qum6sd22rmsewwd75lzlynuve8wz0ah2uw2gltu
+
+Yes, that's correct! This was the first address printed in the transaction dump
+yesterday. If he was able to obtain the same address then surely he will be
+able to move the coins. And so are the children! When they wake up in the
```

Comment:
It's not quite clear to me what "And so are the children" means. Is it "And so will the children" as in "the children will also be able to move the coins"?

Reply:
Thanks, fixed.